packages/graphiql/src/components/GraphiQL.tsx

```diff
@@ -267,19 +280,26 @@ export function GraphiQLInterface(props: GraphiQLInterfaceProps) {
   });
 
   const [activeSecondaryEditor, setActiveSecondaryEditor] = useState<
-    'variables' | 'headers'
+    'variables' | 'extensions' | 'headers'
   >(() => {
     if (
       props.defaultEditorToolsVisibility === 'variables' ||
+      props.defaultEditorToolsVisibility === 'extensions' ||
       props.defaultEditorToolsVisibility === 'headers'
     ) {
       return props.defaultEditorToolsVisibility;
     }
-    return !editorContext.initialVariables &&
-      editorContext.initialHeaders &&
-      isHeadersEditorEnabled
-      ? 'headers'
-      : 'variables';
```

Comment:
I had to remove the ternary and use plain old if statements when introducing the extensions tab (the third tab)

Reply:
Well I guess I could have kept going with nested ternaries but nobody wants to see that 😄